### PR TITLE
Remove ios native dependencies

### PR DESCRIPTION
**Pull request by TsvetanMilanov**
We need to remove our native dependencies - ref and ffi because they are casuing problems with each version of NodeJS which has new version of V8 in it. We are using ref and ffi to communicate with iTunes and execute fs and application management operations on iOS devices. This operations are moved in new dependency - device-lib. The device lib has JavaScript wrapper which we use in the CLI. Everything in the ios device file system and ios device application manager is replaced with the exposed methods from the device lib.

### CODE REVIEW

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:heart:

File: mobile/ios/device/ios-device.ts [hunk 1]
```diff
@@ -1,71 +1,76 @@
...
-		// In case any of the operations had failed, the device status should be Unreachable.
-		if (this.deviceInfo.errorHelp) {
-			this.deviceInfo.status = constants.UNREACHABLE_STATUS;
+	public getApplicationInfo(applicationIdentifier: string): Promise<Mobile.IApplicationInfo> {
+		return this.applicationManager.getApplicationInfo(applicationIdentifier);
+	}
+
+	public async openDeviceLogStream(): Promise<void> {
+		if (this.deviceInfo.status !== constants.UNREACHABLE_STATUS) {
+			await this.$iosDeviceOperations.startDeviceLog(this.deviceInfo.identifier, (data: string) => {
```
Comment (Mitko-Kerezov) [inline mobile/ios/device/ios-device.ts]: There's no need to `await` this - it doesn't return a `Promise`

File: mobile/ios/device/ios-application-manager.ts [hunk 2]
```diff
@@ -173,42 +71,40 @@ export class IOSApplicationManager extends ApplicationManagerBase {
...
-		try {
-			let result = this.$mobileDevice.deviceUninstallApplication(afc, this.$coreFoundation.createCFString(appIdentifier), null, this.uninstallApplicationCallbackPtr);
-			if (result) {
-				this.$errors.failWithoutHelp("AMDeviceUninstallApplication returned '%d'.", result);
-			}
-		} catch (e) {
-			this.$logger.trace(`Error while uninstalling application ${e}.`);
-		}
+		await this.$iosDeviceOperations.uninstall(appIdentifier, [this.device.deviceInfo.identifier], (err: IDeviceOperationError) => {
+			this.$logger.warn(`Failed to uninstall install ${appIdentifier} on device with identifier ${err.deviceIdentifier}`);
```
Comment (rosen-vladimirov) [inline mobile/ios/device/ios-application-manager.ts]: which operation failed - unisntall or install ?

File: commands/device/get-file.ts [hunk 3]
```diff
@@ -1,15 +1,25 @@
...
+		private $project: Project.IProjectBase,
+		private $errors: IErrors,
 		private $options: ICommonOptions) { }
 
-	allowedParameters: ICommandParameter[] = [this.$stringParameter];
+	public allowedParameters: ICommandParameter[] = [this.$stringParameter, this.$stringParameter];
 
 	public async execute(args: string[]): Promise<void> {
 		await this.$devicesService.initialize({ deviceId: this.$options.device, skipInferPlatform: true });
+		let appIdentifier = args[1];
```
Comment (rosen-vladimirov) [inline commands/device/get-file.ts]: as we modify the command, shouldn't we delete the `$options.app` that has been used before (not in the command itself, but in the service behind it. If I understand correctly, we'll support getting of file only from application sandbox, so the `$options.app` is a mandatory argument now. So... let's remove `$options.app`

File: mobile/ios/device/ios-application-manager.ts [hunk 4]
```diff
@@ -173,42 +71,40 @@ export class IOSApplicationManager extends ApplicationManagerBase {
...
+		});
 
 		this.$logger.trace("Application %s has been uninstalled successfully.", appIdentifier);
 	}
 
 	public async startApplication(appIdentifier: string): Promise<void> {
 		if (this.$hostInfo.isWindows && !this.$staticConfig.enableDeviceRunCommandOnWindows) {
-			this.$errors.fail("$%s device run command is not supported on Windows for iOS devices.", this.$staticConfig.CLIENT_NAME.toLowerCase());
+			// this.$errors.fail("$%s device run command is not supported on Windows for iOS devices.", this.$staticConfig.CLIENT_NAME.toLowerCase());
```
Comment (rosen-vladimirov) [inline mobile/ios/device/ios-application-manager.ts]: I believe we need it.

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by Plamen5kov** [comment]
related issue: https://github.com/NativeScript/nativescript-cli/issues/2562

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by justcodebuilduser** [comment]
:broken_heart:

**Comment by TsvetanMilanov** [comment]
run ci

**Comment by justcodebuilduser** [comment]
:heart:

**Comment by justcodebuilduser** [comment]
:heart:

**Pull request merged**